lib/nokogiri/css/parser.rb

```diff
@@ -8,9 +8,15 @@
 
 
 require 'nokogiri/css/parser_extras'
+
 module Nokogiri
   module CSS
     class Parser < Racc::Parser
+
+
+def unescape_css_identifier(identifier)
+  identifier.gsub(/\\(?:([^0-9a-fA-F])|([0-9a-fA-F]{1,6})\s?)/){ |m| $1 || [$2.hex].pack('U') }
```

Comment:
and from http://www.w3.org/TR/css3-selectors/#selector-syntax 
"Characters in Selectors can be escaped with a backslash according to the same escaping rules as CSS. [CSS21]"